### PR TITLE
Give reasonable error messages when trying to use p::d::Triangulation without p4est

### LINKED ISSUE #10807: Can't compile p::d::T class when p4est not detected
We have a sham class for `p::d::T` that is selected when p4est was not detected: It doesn't do anything useful, but allows codes to compile. But that doesn't work right now because there are unimplemented virtual functions:
```
/home/shyaan/dealii/include/deal.II/distributed/tria.h:1402:11: note:   because the following virtual functions are pure within ‘dealii::parallel::distributed::Triangulation<3, 3>’:
     class Triangulation
           ^~~~~~~~~~~~~
In file included from /home/shyaan/dealii/include/deal.II/dofs/dof_handler.h:29:0,
                 from /home/shyaan/dealii/include/deal.II/numerics/vector_tools_boundary.h:22,
                 from /home/shyaan/dealii/include/deal.II/numerics/vector_tools.h:22,
                 from /home/shyaan/thm/./include/boundaryvalues.h:8,
                 from /home/shyaan/thm/src/main.cc:1:
/home/shyaan/dealii/include/deal.II/distributed/tria_base.h:105:5: note:        bool dealii::parallel::TriangulationBase<dim, spacedim>::is_multilevel_hierarchy_constructed() const [with int dim = 3; int spacedim = 3]
     is_multilevel_hierarchy_constructed() const = 0;
```

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes #10807. Also, match the constructor of the "real" implementation.

### CODE REVIEW

File: include/deal.II/distributed/tria.h [hunk 1]
```diff
@@ -957,11 +957,61 @@ namespace parallel
...
+      {
+        return false;
+      }
+
+      /**
+       * Dummy replacement to allow for better error messages when compiling
+       * this class.
+       */
+      virtual void
+      save(const std::string &filename) const override
```
Comment (peterrum) [inline include/deal.II/distributed/tria.h:996]: unsued variable?

File: include/deal.II/distributed/tria.h [hunk 2]
```diff
@@ -957,11 +957,61 @@ namespace parallel
...
+      {}
+
+      /**
+       * Dummy replacement to allow for better error messages when compiling
+       * this class.
+       */
+      virtual void
+      load(const std::string &filename,
+           const bool         autopartition = true) override
```
Comment (peterrum) [inline include/deal.II/distributed/tria.h:1005]: Same here?

**Review by peterrum — APPROVED** [review]
Thanks!

**Comment by bangerth** [comment]
OK if you fix the warnings about the unused function arguments.

File: include/deal.II/distributed/tria.h (hunk 1 above)
Comment (masterleinad) [reply include/deal.II/distributed/tria.h]: Fixed.

File: include/deal.II/distributed/tria.h (hunk 2 above)
Comment (masterleinad) [reply include/deal.II/distributed/tria.h]: Fixed.

**Pull request merged**